prescient.el

```diff
@@ -327,47 +347,65 @@ as a sub-query delimiter."
       (format "\\(%s\\)" regexp)
     regexp))
 
-(defun prescient-literal-regexp (query &optional with-groups)
+(defun prescient-literal-regexp (query props)
```

Comment:
```suggestion
(cl-defun prescient-literal-regexp (query &key with-groups &allow-other-keys)
```

So it turns out that `cl-lib` has a neat way of doing this plist pattern out of the box. You just list the properties of the plist (without the leading colons), and they are automatically destructured if present, otherwise they default to nil. You have to include `&allow-other-keys` for forward compatibility.

In terms of the interface, this basically is the same as what you've already implemented, except that the `props` is a variadic argument (`&rest props`). In the docstring for `prescient-filter-alist`, we'd advise using the function signature with `cl-defun` and `&allow-other-keys`.